.github/workflows/build_aarch64.yml

```diff
@@ -0,0 +1,315 @@
+#
+# Copyright (c) 2014-present, The osquery authors
+#
+# This source code is licensed as defined by the LICENSE file found in the
+# root directory of this source tree.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR GPL-2.0-only)
+#
+
+# Due to a limitation in how GitHub Actions works, we can't reference
+# jobs in another file inside the `needs` statement.
+#
+# This configuration file takes care of Linux on aarch64.
+name: build_aarch64
+
+on:
+  push:
+    branches:
+      - 'master'
+
+    tags:
+      - '*'
+
+# If the initial code sanity checks are passing, then one job
+# per [`platform` * `build_type`] will start, building osquery
+# and generating packages that are later attached to the commit
+# (or PR) as build artifacts.
+jobs:
+  # Starts optional self-hosted runners.
+  start_aarch64_ec2_runners:
+    runs-on: ubuntu-latest
+    outputs:
+      label: ${{ steps.start_ec2_runner.outputs.label }}
+      ec2-instance-id: ${{ steps.start_ec2_runner.outputs.ec2-instance-id }}
+    steps:
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.EC2_GITHUB_RUNNER_AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.EC2_GITHUB_RUNNER_AWS_SECRET_ACCESS_KEY }}
+          aws-region: ${{ secrets.EC2_GITHUB_RUNNER_AWS_REGION }}
+      - name: Start aarch64 EC2 runner
+        id: start_ec2_runner
+        uses: theopolis/ec2-github-runner@main
+        with:
+          mode: start
+          github-token: ${{ secrets.EC2_GITHUB_RUNNER_GH_PERSONAL_ACCESS_TOKEN }}
+          ec2-image-id: ami-063d99d9ce1bebbe8
+          ec2-instance-type: r6g.4xlarge
+          subnet-id: subnet-06390365e72579736
+          security-group-id: sg-0757a3162c999331b
+
+  # Stops optional self-hosted runners.
+  stop_aarch64_ec2_runners:
+    needs:
+      - start_aarch64_ec2_runners
+      - build_linux_aarch64
```

Comment:
As I understand it, `needs` requires the example jobs to be successful. So I'm not sure this cleanup will trigger if the build fails. 

I can't quickly tell from the docs if `continue-on-error` will solve this, without also making the job allow the whole thing to pass.

Reply:
Good call, I think the solution is to add an if: always().